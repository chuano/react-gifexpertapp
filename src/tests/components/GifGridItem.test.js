import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';

describe("GifGridItem test", () => {

  const title = 'Un título'
  const url = 'http://localhost/algo.jpg'
  const wrapper = shallow(<GifGridItem title={ title } url={ url }/>)

  test("debe mostrar correctamente el componente", () => {
    expect(wrapper).toMatchSnapshot()
  });

  test('debe tener un párrafo con el title', () => {
    const p = wrapper.find('p')
    expect(p.text().trim()).toBe(title)
  })
  
  test('debe tener src y alt', () => {
    const img = wrapper.find('img')
    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  })

  test('debe tener animated__fadeIn', () => {
    const div = wrapper.find("div");
    const className = div.prop('className')
    expect(className.includes("animate__fadeIn")).toBe(true);
  })
});
