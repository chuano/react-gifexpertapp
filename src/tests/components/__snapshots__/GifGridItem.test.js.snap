// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem test debe mostrar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p>
    Un título
  </p>
  <img
    alt="Un título"
    src="http://localhost/algo.jpg"
  />
</div>
`;
